server/src/test/scala/cromwell/jobstore/JobStoreWriterSpec.scala

```diff
@@ -63,53 +62,59 @@ class JobStoreWriterSpec extends CromwellTestKitWordSpec with Matchers with Befo
 
   "JobStoreWriter" should {
     "be able to collapse writes together if they arrive while a database access is ongoing" in {
+      runWithDatabase(databaseConfig) { workflowStore =>
+        val jobStoreWriter = TestFSMRef(new JobStoreWriterActor(database, 5, flushFrequency, TestProbe().ref, 1000, access(workflowStore)))
 
-      // Send a job completion. The database will hang.
-      sendRegisterCompletion(1)
-      val writer = jobStoreWriter.underlyingActor
-      eventually {
-        writer.stateName shouldBe Processing
-      }
+        // Send a job completion. The database will hang.
+        sendRegisterCompletion(jobStoreWriter)(1)
+        val writer = jobStoreWriter.underlyingActor
+        eventually {
+          writer.stateName shouldBe Processing
+        }
 
-      // Send some more completions.  These should pile up in the state data.
-      List(2, 3) foreach sendRegisterCompletion
+        // Send some more completions.  These should pile up in the state data.
+        List(2, 3) foreach sendRegisterCompletion(jobStoreWriter)
 
-      writer.stateData.weight should equal(2)
-      database.continue()
+        writer.stateData.weight should equal(2)
+        database.continue()
 
-      assertReceived(expectedJobStoreWriteAcks = 3)
+        assertReceived(expectedJobStoreWriteAcks = 3)
 
-      assertDb(
-        totalWritesCalled = 2,
-        jobCompletionsRecorded = 3,
-        workflowCompletionsRecorded = 0
-      )
+        assertDb(
+          totalWritesCalled = 2,
+          jobCompletionsRecorded = 3,
+          workflowCompletionsRecorded = 0
+        )
+      }
     }
 
     "be able to skip job-completion writes if the workflow completes, but still respond appropriately" in {
-
-      // Send a job completion. The database will hang.
-      sendRegisterCompletion(1)
-      val writer = jobStoreWriter.underlyingActor
-      eventually {
-        writer.stateName shouldBe Processing
+      runWithDatabase(databaseConfig) { workflowStore =>
+        val jobStoreWriter = TestFSMRef(new JobStoreWriterActor(database, 5, flushFrequency, TestProbe().ref, 1000, access(workflowStore)))
+
+        // Send a job completion. The database will hang.
+        sendRegisterCompletion(jobStoreWriter)(1)
+        val writer = jobStoreWriter.underlyingActor
+        eventually {
+          writer.stateName shouldBe Processing
+        }
+
+        // Send some more completions.  These should pile up in the state data.
+        List(2, 3) foreach sendRegisterCompletion(jobStoreWriter)
+        jobStoreWriter ! RegisterWorkflowCompleted(workflowId)
+
+        writer.stateData.weight should equal(3)
+        // The testing DB intentionally blocks after the first write until `continue` is called.
+        database.continue()
+
+        assertReceived(expectedJobStoreWriteAcks = 3)
+
+        assertDb(
+          totalWritesCalled = 2,
+          jobCompletionsRecorded = 1,
+          workflowCompletionsRecorded = 1
+        )
```

Comment:
This file also appears to not be much changed?

Reply:
Same reason as above.